packages/functions/test/unit/cache/index.test.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  afterEach,
+  beforeEach,
+  describe,
+  expect,
+  Mock,
+  test,
+  vitest,
+} from 'vitest';
+
+import { getFunctionCache } from '../../../src/cache/index';
+import { InMemoryCache } from '../../../src/cache/in-memory-cache';
+import { getContext } from '../../../src/get-context';
+
+vitest.mock('../../../src/get-context', () => ({
+  getContext: vitest.fn(),
+}));
+
+describe('getRuntimeCache', () => {
+  let mockCache: InMemoryCache;
+
+  beforeEach(() => {
+    mockCache = new InMemoryCache();
+    (getContext as Mock).mockReturnValue({ cache: mockCache });
+  });
+
+  afterEach(() => {
+    vitest.clearAllMocks();
+  });
+
+  test('should use the context cache if available', async () => {
+    const cache = await getFunctionCache();
```

Comment:
super nit: Perhaps `getFunctionCache` doesn't return a promise?

```suggestion
    const cache = getFunctionCache();
```

Reply:
good catch - missed that when I changed getFunctionCache() to not return a promise